components/paystack/actions/initialize-transaction.mjs

```diff
@@ -0,0 +1,46 @@
+import paystack from "../paystack.app.mjs";
+
+export default {
+  key: "paystack-initialize-transaction",
+  name: "Initialize Transaction",
+  description:
+    "Initializes a new transaction on Paystack. [See the documentation](https://paystack.com/docs/api/transaction/#initialize)",
+  version: "0.0.10",
+  type: "action",
+  props: {
+    paystack,
+    email: {
+      propDefinition: [paystack, "email"],
+      optional: false,
+    },
+    amount: {
+      propDefinition: [paystack, "amount"],
+      optional: false,
+    },
+    currency: {
+      propDefinition: [paystack, "currency"],
+      optional: true,
+    },
+    reference: {
+      propDefinition: [paystack, "reference"],
+      optional: true,
+    },
+    callback_url: {
+      propDefinition: [paystack, "callback_url"],
+      optional: true,
+    },
+    metadata: {
+      propDefinition: [paystack, "metadata"],
+      optional: true,
+    },
+  },
+  async run({ $ }) {
+    const response = await this.paystack.initializeTransaction({
+      email: this.email,
+      amount: this.amount,
+    });
```

Comment:
```suggestion
    const response = await this.paystack.initializeTransaction({
      data: {
        email: this.email,
        amount: this.amount,
      }
    });
```

Reply:
Got no problem with this change, but just curious if this is the preferred way to pass args? Nesting the data object inside of another object seems a little redundant